app/src/main/java/protect/card_locker/ScanActivity.java

```diff
@@ -116,6 +124,8 @@ public void possibleResultPoints(List<ResultPoint> resultPoints) {
     protected void onResume() {
         super.onResume();
         capture.onResume();
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED)
+            showCameraPermissionMissingText(true);
```

Comment:
Oh yeah the logic is inverted here, but I will fix it.
Also, I tested it on my Android phone(which runs Android 11) and when I un-granted the camera permission and switched back to the app, the system permission inquiry dialog will pop up. When I deny it, the camera permission denial UI will show up. Since I put the checking code in onResume(), according to Android's Activity lifecycle, it should be executed when switching back to the app.